lib/index.js

```diff
@@ -115,6 +115,42 @@ const resourcesProperties = {
   extent: mappingTemplates.exactString,
   formerTitle: mappingTemplates.fulltextFolded,
   genreForm: mappingTemplates.fulltextWithRawFolded,
+  holdings: {
+    type: 'nested',
+    properties: {
+      checkInBoxes: {
+        type: 'nested',
+        properties: {
+          copies: mappingTemplates.number,
+          coverage: mappingTemplates.exactString,
+          position: mappingTemplates.number,
+          shelfMark: mappingTemplates.exactStringNotIndexed,
+          status: mappingTemplates.exactString,
+          type: mappingTemplates.exactString
+        }
+      },
+      format: mappingTemplates.exactString,
+      holdingStatement: mappingTemplates.exactString,
+      identifier: mappingTemplates.exactString,
+      identifierV2: {
+        properties: {
+          value: mappingTemplates.exactString,
+          type: mappingTemplates.exactString,
+          identifierStatus: mappingTemplates.exactString
+        }
+      },
```

Comment:
For the `items` mapping, I originally only had the `identifier` property - not `identifierV2`. The `identifier` property just stores `keyword` typed strings like "urn:barcode:1234". That's lazy json-ld (and makes keyword matching on the value difficult) so I later wanted something more robust and added `identifierV2`, which is a proper object like `{ value: '1234', type: 'bf:Barcode' }`.. If I could have I would have modified the mapping type for `identifier`. Because doing that would require building a brand new index, I just added `identifierV2` and [handle stripping the "V{N}" in the serializer](https://github.com/NYPL-discovery/discovery-api/blob/e96f7814a4b277324ba4f6095b00c77eca422759/lib/jsonld_serializers.js#L100). It's not perfect, but it's a general solution for radically changing property mappings iteratively.

_I say all this because_ (and from your close reading of the repo you may know all this already) I see you're adding `identifier` and `identifierV2` straight away for the holding object. Do you anticipate needing the "urn-style" format of holding identifiers? We actually do still use the urn-style identifiers [for](https://github.com/NYPL-discovery/discovery-api/blob/835862e6313bb805b39f30d639fa0f19ddbfcfd2/lib/delivery-locations-resolver.js#L80) [some](https://github.com/NYPL-discovery/discovery-api/blob/835862e6313bb805b39f30d639fa0f19ddbfcfd2/lib/delivery-locations-resolver.js#L155) [filters](https://github.com/NYPL-discovery/discovery-api/blob/835862e6313bb805b39f30d639fa0f19ddbfcfd2/lib/delivery-locations-resolver.js#L165) but I think the practice can be deprecated. There may be a reason to store identifiers in the two different formats for holdings, but I thought providing some background on the `identifier`/`identifierV2` _may_ cause you to reconsider..

Reply:
Thanks Paul, it does. I was attempting to cover my bases in case the typed strings in the `identifier` field were necessary in some part of the API that I wasn't familiar with. But since that's not the case, I'll just create the identifier objects (I certainly don't need them for anything we're doing with the holding records. There actually should be very few if any reasons to need to look them up directly).